File: java-testapp/src/main/java/com/fsryan/tools/logging/testappjava/SysOutDevMetricsLogger.java
package com.fsryan.tools.logging.testappjava;

import com.fsryan.tools.logging.FSDevMetricsLogger;

import java.util.HashMap;
import java.util.Map;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public class SysOutDevMetricsLogger implements FSDevMetricsLogger {
    
    @Nonnull
    @Override
    public String id() {
        return "sysout";
    }

    @Override
    public void alarm(@Nonnull Throwable t, @Nonnull Map<String, String> attrs) {
        System.out.println("[FS/ALARM]: " + t + "; attrs = " + attrs);
    }

    @Override
    public void watch(@Nonnull String msg, @Nonnull Map<String, String> attrs) {
        System.out.println("[FS/WATCH]: " + "msg = " + msg + "; attrs = " + attrs);
    }

    @Override
    public void info(@Nonnull String msg, @Nonnull Map<String, String> attrs) {
        System.out.println("[FS/INFO]: " + "msg = " + msg + "; attrs = " + attrs);
    }

    @Override
    public void metric(@Nonnull String operationName, long durationNanos) {
        System.out.println("[FS/METRIC]: '" + operationName + "' took " + durationNanos + " nanos");
    }

    @Override
    public boolean runInTestEnvironment() {
        return false;
    }
}

File: kotlin-testapp/src/main/java/com/fsryan/tools/logging/testappkt/SysOutDevMetricsLogger.kt
package com.fsryan.tools.logging.testappkt

import com.fsryan.tools.logging.FSDevMetricsLogger

internal class SysOutDevMetricsLogger : FSDevMetricsLogger {
    override fun id(): String = "sysout"

    override fun alarm(t: Throwable, attrs: Map<String, String>) = println("[FS/ALARM]: $t; attrs: $attrs")

    override fun watch(msg: String, attrs: Map<String, String>) = println(
        "[FS/WATCH]: msg: $msg; attrs: $attrs}"
    )

    override fun info(msg: String, attrs: Map<String, String>) = println(
        "[FS/INFO]: msg: $msg; attrs: $attrs"
    )

    override fun metric(operationName: String, durationNanos: Long) = println("[FS/METRIC]: '$operationName' took $durationNanos nanos")
}